processor/processor.go

```diff
@@ -989,6 +964,26 @@ func (proc *HandleT) processJobsForDest(jobList []*jobsdb.JobT, parsedEventList
 	proc.pStatsDBW.Print()
 }
 
+func getTruncatedEventList(jobList []*jobsdb.JobT, maxEvents int) (truncatedList []*jobsdb.JobT, totalEvents int, gotMaxEvents bool) {
```

Comment:
1. Do we need gotMaxEvents 
2. Can we do with a single return? Makes it more readable in this case.
```
	lastIndex := len(jobList)

  for idx, job := range jobList {
		eventsInJob := len(gjson.GetBytes(job.EventPayload, "batch").Array())
		totalEvents += eventsInJob
		if totalEvents >= maxEvents {
			// return atleast one job to prevent processing first job with events > maxEvents
			// eg. first job has 1500 events and maxEvents is only 1000
			if idx == 0 {
				lastIndex = 1
			} else if totalEvents == maxEvents {
				// if both are equal, include current index of slice also
				lastIndex = idx + 1
			} else {
                lastIndex = idx
                totalEvents = totalEvents - eventsInJob
            }
            break
		 }
	}

  return jobList[:lastIndex], totalEvents
```